worker-message-prioritization-rerouting/src/main/java/com/github/workerframework/workermessageprioritization/rerouting/MessageRouter.java

```diff
@@ -113,14 +113,47 @@ public void route(final Document document) {
         }
 
     }
-    
-    private boolean shouldReroute(final ResponseQueue successQueue) {
+
+    public String route(final String originalQueueName, final String tenantId) {
+
+        final Queue originalQueue;
+        try {
+            originalQueue = queuesCache.get(originalQueueName);
+        } catch (final ExecutionException e) {
+            LOGGER.error(
+                    "Unable to retrieve the definition of original queue '{}', reverting to original queue. {}",
+                    originalQueueName, e);
+
+            return originalQueueName;
+        }
+
+        if(shouldReroute(originalQueueName)) {
+
+            final String successQueueName = originalQueueName + LOAD_BALANCED_INDICATOR + "/" + tenantId;
```

Comment:
Move the body of this if statement into a private "route" method so it's not duplicated between the two public route methods

Reply:
I'm not sure how to extract anything out here as the if blocks are different.